templates.rst

```diff
@@ -0,0 +1,1061 @@
+(e.g. ``<a href="{{ url('blog_index') }}"> ... </a>``).
+
+.. _templates-link-to-assets:
+
+Linking to CSS, JavaScript and Image Assets
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Templates usually link to images, JavaScript, stylesheets and other assets.
+Instead of writing the asset URLs by hand, use the ``asset()`` function to
+generate the URLs.
+
+To use this function, install the *asset* package:
+
+.. code-block:: terminal
+
+    $ composer require symfony/asset
+
+You can now use the ``asset()`` function:
+
+.. code-block:: html+twig
+
+    <img src="{{ asset('images/logo.png') }}" alt="Symfony!"/>
+
+    <link href="{{ asset('css/blog.css') }}" rel="stylesheet"/>
+
+    <script src="{{ asset('bundles/acme/js/loader.js') }}"></script>
+
+The ``asset()`` function's main purpose is to make your application more portable.
+If your application lives at the root of your host (e.g. ``https://example.com``),
+then the rendered paths should be ``/images/logo.png``. But if your application
+lives in a subdirectory (e.g. ``https://example.com/my_app``), each asset path
+should render with the subdirectory (e.g. ``/my_app/images/logo.png``). The
+``asset()`` function takes care of this by determining how your application is
+being used and generating the correct paths accordingly.
+
+.. tip::
+
+    The ``asset()`` function supports various cache busting techniques via the
+    :ref:`version <reference-framework-assets-version>`,
+    :ref:`version_format <reference-assets-version-format>`, and
+    :ref:`json_manifest_path <reference-assets-json-manifest-path>` configuration options.
+
+If you need absolute URLs for assets, use the ``absolute_url()`` Twig function
+as follows:
+
+.. code-block:: html+twig
+
+    <img src="{{ absolute_url(asset('images/logo.png')) }}" alt="Symfony!"/>
+
+.. tip::
+
+    Read the :doc:`Symfony's Webpack Encore documentation </frontend>` to learn
+    about integrating modern asset building features into Symfony applications.
+
+.. _twig-app-variable:
+
+The App Global Variable
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony creates for each request a context variable which is injected in every
+Twig template automatically as a variable called ``app``. It provides access to
+some application information:
+
+.. code-block:: html+twig
+
+    <p>Username: {{ app.user.username ?? 'Anonymous user' }}</p>
+    {% if app.debug %}
+        <p>Request method: {{ app.request.method }}</p>
+        <p>Application Environment: {{ app.environment }}</p>
+    {% endif %}
+
+The ``app`` variable (which is an instance of :class:`Symfony\\Bridge\\Twig\\AppVariable`)
+gives you access to these variables:
+
+``app.user``
+    The representation of the :ref:`current user <create-user-class>` or ``null``
+    if there is none. The value stored in this variable can be a
+    :class:`Symfony\\Component\\Security\\Core\\User\\UserInterface` object, any
+    other object which implements a ``__toString()`` method or even a regular string.
```

Comment:
I think we should completely stop supporting anyone who is setting the user to anything other than a User object.

```rst
The :ref:`current user object <create-user-class>` or ``null`` if
the user is not authenticated.
```

And that's it.